### PR TITLE
autotest: Improve Landing-Drift

### PR DESCRIPTION
**Pull request by samuelctabor**
Now checks that an aborted landing occurs.

### CODE REVIEW

File: Tools/autotest/arduplane.py [hunk 1]
```diff
@@ -3291,7 +3291,23 @@ def fly_landing_baro_drift(self):
         self.wait_ready_to_arm()
         self.arm_vehicle()
 
-        self.fly_mission("ap-circuit.txt", mission_timeout=1200)
+        # Load and start mission
+        self.load_mission("ap-circuit.txt", strict=True)-1
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:3295]:
```suggestion
        self.load_mission("ap-circuit.txt", strict=True)
```

File: Tools/autotest/arduplane.py [hunk 2]
```diff
@@ -3291,7 +3291,23 @@ def fly_landing_baro_drift(self):
...
+        self.set_current_waypoint(0, check_afterwards=False)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:3296]: Why not checking afterwards?

File: Tools/autotest/arduplane.py [hunk 3]
```diff
@@ -3291,7 +3291,23 @@ def fly_landing_baro_drift(self):
...
+        self.change_mode('AUTO')
+        self.wait_current_waypoint(1, timeout=5)
+        self.wait_groundspeed(0, 30, timeout=5)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:3299]: Perhaps you should constrain this wait a bit more?  "anywhere between 0 and 30" is pretty wide?

File: Tools/autotest/arduplane.py (hunk 2 above)
Comment (samuelctabor) [reply Tools/autotest/arduplane.py:3296]: I actually copied the lines from fly_mission, I can try adding the check

**Comment by samuelctabor** [comment]
Updated to check the WP after setting (and changed to setting WP1 rather than WP0) and reduced groundspeed range.

**Comment by samuelctabor** [comment]
Any further comments?

File: Tools/autotest/arduplane.py [hunk 4]
```diff
@@ -3291,7 +3291,23 @@ def fly_landing_baro_drift(self):
...
+        # Wait for landing waypoint
+        self.wait_current_waypoint(9, timeout=1200)
+
+        # Wait for landing restart
+        self.wait_current_waypoint(5, timeout=60)
+
+        # Wait for landing waypoint (second attempt)
+        self.wait_current_waypoint(9, timeout=1200)
+
+        self.wait_statustext("Auto disarmed", timeout=120)
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:3310]: This is probably better as `wait_disarmed()`

**Review by peterbarker — APPROVED** [review]
Looks good.

It would be nice if we tested a bit more about what we're doing in this flight, but this is certainly an improvement (and provides nice repeatable flights).

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
I'm not an autotest expert but the Landing-Drift Plane autotest seems to be failing a lot recently.  I don't want to complain too much but flapping tests cost developers a significant amount of time because they need to keep checking the CI and restarting failed jobs.

**Pull request merged**